core/templates/pages/collection-player-page/collection-player-page.component.ts

```diff
@@ -0,0 +1,397 @@
+        let x = (i % 2) ? 30 : 470;
+        sParameterExtension += x + ' ' + this.y + ', ';
+        this.y += 20;
+        sParameterExtension += 250 + ' ' + this.y + ', ';
+        this.y += 200;
+      }
+      if (sParameterExtension !== '') {
+        this.pathSvgParameters += ' S ' + sParameterExtension;
+      }
+    }
+    if (collectionNodeCount % 2 === 0) {
+      if (collectionNodeCount === 2) {
+        this.svgHeight = this.MIN_HEIGHT_FOR_PATH_SVG_PX;
+      } else {
+        this.svgHeight = this.y - this.EVEN_SVG_HEIGHT_OFFSET_PX;
+      }
+    } else {
+      if (collectionNodeCount === 1) {
+        this.svgHeight = this.MIN_HEIGHT_FOR_PATH_SVG_PX;
+      } else {
+        this.svgHeight = this.y - this.ODD_SVG_HEIGHT_OFFSET_PX;
+      }
+    }
+  }
+
+  generatePathIconParameters(): IconParametersArray[] {
+    let collectionNodes = this.collection.getCollectionNodes();
+    let iconParametersArray = [];
+    iconParametersArray.push({
+      thumbnailIconUrl:
+        collectionNodes[0].getExplorationSummaryObject(
+        ).thumbnail_icon_url.replace('subjects', 'inverted_subjects'),
+      left: '225px',
+      top: '35px',
+      thumbnailBgColor:
+        collectionNodes[0].getExplorationSummaryObject(
+        ).thumbnail_bg_color
+    });
+
+    // Here x and y represent the co-ordinates for the icons in the
+    // path.
+    let x = this.ICON_X_MIDDLE_PX;
+    let y = this.ICON_Y_INITIAL_PX;
+    let countMiddleIcon = 1;
+
+    for (let i = 1; i < this.collection.getCollectionNodeCount(); i++) {
+      if (countMiddleIcon === 0 && x === this.ICON_X_MIDDLE_PX) {
+        x = this.ICON_X_LEFT_PX;
+        y += this.ICON_Y_INCREMENT_PX;
+        countMiddleIcon = 1;
+      } else if (countMiddleIcon === 1 && x === this.ICON_X_MIDDLE_PX) {
+        x = this.ICON_X_RIGHT_PX;
+        y += this.ICON_Y_INCREMENT_PX;
+        countMiddleIcon = 0;
+      } else {
+        x = this.ICON_X_MIDDLE_PX;
+        y += this.ICON_Y_INCREMENT_PX;
+      }
+      iconParametersArray.push({
+        thumbnailIconUrl:
+          collectionNodes[i].getExplorationSummaryObject(
+          ).thumbnail_icon_url.replace(
+            'subjects', 'inverted_subjects'),
+        left: x + 'px',
+        top: y + 'px',
+        thumbnailBgColor:
+          collectionNodes[i].getExplorationSummaryObject(
+          ).thumbnail_bg_color
+      });
+    }
+    return iconParametersArray;
+  }
+
+  getExplorationUrl(explorationId: string): string {
+    return (
+      '/explore/' + explorationId + '?collection_id=' +
+      this.collectionId);
+  }
+
+  getExplorationTitlePosition(index: number): string {
+    if (index % 2 === 0) {
+      return '8px';
+    } else if ((index + 1) % 2 === 0 && (index + 1) % 4 !== 0) {
+      return '30px';
+    } else if ((index + 1) % 4 === 0) {
+      return '-40px';
+    }
+  }
+
+  scrollToLocation(id: string): void {
+    this.windowRef.nativeWindow.location.hash = '#' + (id);
```

Comment:
Can you add a video showing that this works?

Reply:
Animations are applied on this selector `oppia-collection-animate-slide`. This selector doesn't exist in the corresponding template. I haven't worked on animations much, correct me if that's not the case. For scrolling I've updated the video in description